src/main/java/com/rk/rentacarapp/services/impl/VehicleServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.rk.rentacarapp.services.impl;
+
+import com.rk.rentacarapp.model.Vehicle;
+import com.rk.rentacarapp.model.VehicleType;
+import com.rk.rentacarapp.repository.VehicleRepository;
+import com.rk.rentacarapp.services.VehicleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class VehicleServiceImpl implements VehicleService {
+
+    @Autowired
+    private VehicleRepository vehicleRepository;
+
+    public List<Vehicle> findAllVehicles(){
+        return vehicleRepository.findAll();
+    }
+
+    public Vehicle findVehicleById(Long id){
+        return vehicleRepository.findById(id).get();
+    }
+
+    public Vehicle findVehicleByRegPlate(String regPlate){
+        return vehicleRepository.findByRegPlate(regPlate);
+    }
+
+    public List<Vehicle> findAllVehiclesByType(VehicleType vehicleType){
+        return vehicleRepository.findAllByVehicleType(vehicleType);
+    }
+
+    @Override
+    public List<Vehicle> findTop3(VehicleType vehicleType) {
+        return vehicleRepository.findTop3ByVehicleType(vehicleType);
+    }
+
+    @Override
+    public Vehicle addNewVehicle(Vehicle vehicle) {
+        return vehicleRepository.save(vehicle);
+    }
+
+    @Override
+    public Vehicle patchVehicle(Long id, Vehicle vehicle) {
+        Vehicle v = vehicleRepository.findById(id).get();
```

Comment:
Clean code (you shouldn't use single letter variable names)

So this should be a patchedVehicle if I'm getting your code right